regparser/notice/util.py

```diff
@@ -6,21 +6,29 @@ def prepost_pend_spaces(el):
     them. Account for this by adding spaces around the el where needed."""
     not_append_space = """@#$(-'" \t\n"""
     not_prepend_space = """%):?!,. \t\n"""
-    space_added = False
 
+    non_empty = el.text or el.getchildren()
     parent = el.getparent()
     prev = el.getprevious()
-    if prev is not None:
-        if prev.tail and prev.tail[-1] not in not_append_space:
+
+    prev_text = (prev is not None and prev.tail) or parent.text
+    if non_empty:
```

Comment:
The following block of code is really hard to read. What can we do to make it more readable?